app/src/main/java/com/example/anfcodetest/MainActivity.kt

```diff
@@ -1,55 +1,36 @@
 package com.example.anfcodetest
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
 import androidx.recyclerview.widget.LinearLayoutManager
-import com.fasterxml.jackson.module.kotlin.*
+import com.hannesdorfmann.mosby3.mvp.MvpActivity
 import kotlinx.android.synthetic.main.activity_main.*
-import okhttp3.*
-import java.io.IOException
 
-val url = "https://www.abercrombie.com/anf/nativeapp/qa/codetest/codeTest_exploreData.json"
-
-class MainActivity : AppCompatActivity(), MainContract.View {
+class MainActivity : MvpActivity<MainContract.View, MainContract.Presenter>(), MainContract.View {
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     setContentView(R.layout.activity_main)
-
     loadCards()
   }
 
+  override fun displayCards(cards: List<Card>) {
+    val adapter = MainAdapter(cards)
+    runOnUiThread {
+      recyclerView_main.adapter = adapter
+      adapter.notifyDataSetChanged()
+    }
+  }
+
   override fun loadCards() {
-    fetchJson(url)
+    getPresenter().fetchJson(PAGE_DATA_URL)
```

Comment:
I think you could also use just `presenter` here to access the presenter and fetch the JSON!